enhancements/storage/csi-secrets-store.md

```diff
@@ -0,0 +1,253 @@
+
+The user experience of needing to install a separate provider plugin after the
+operator is installed is not ideal. However, it is not possible for the operator
+to reliably derive which provider should be used based on the chosen platform.
+It is valid use case to install the Hashicorp Vault provider on a cluster that
+is running on Azure or AWS for example. Therefore, the cluster admin must decide
+which provider should be supported in their environment.
+
+### Drawbacks
+
+See [Risks and Mitigations](#risks-and-mitigations).
+
+## Design Details
+
+### Inline Ephemeral Volumes
+
+This driver is different from other CSI drivers in that it supports only
+[ephemeral volumes](https://github.com/openshift/secrets-store-csi-driver-operator/blob/d271705b2e56b20581dcc04514c9b307b67fbeec/assets/csidriver.yaml#L14)
+and not persistent volumes. By design, the lifecycle of these volumes is the same
+as the pod itself: the volume gets deleted when the pod gets deleted.
+As such, this driver does not have a controller, since there is no provisioning
+of persistent volumes through this driver. Only a node DaemonSet is created to support
+mounting ephemeral volumes on the node where the pod runs.
+
+See more on inline ephemeral volumes in the
+[OpenShift docs](https://docs.openshift.com/container-platform/4.13/storage/container_storage_interface/ephemeral-storage-csi-inline.html)
+and the
+[CSI Inline Ephemeral Volume Security enhancement](https://github.com/openshift/enhancements/blob/b12cefd0f3fa438ec08f84f19c0934a35efa1877/enhancements/storage/csi-inline-vol-security.md).
+
+The operator will create the `secrets-store.csi.k8s.io` CSIDriver object with
+`security.openshift.io/csi-ephemeral-volume-profile: "restricted"` by default
+(see the manifest [here](https://github.com/openshift/secrets-store-csi-driver-operator/blob/d271705b2e56b20581dcc04514c9b307b67fbeec/assets/csidriver.yaml#L8)) so that
+the driver can be used from `restricted` namespaces by default. The cluster admin
+can change this value if they wish to limit the use of this driver to `privileged`
+or `baseline` namespaces.
+
+### Publishing New Provider Plugins
+
+Refer to [upstream guidance for provider implementation](https://secrets-store-csi-driver.sigs.k8s.io/providers.html) when developing a new provider plugin. Each provider is expected to:
+- Use the functions and data structures in [service.pb.go](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/provider/v1alpha1/service.pb.go) to implement the server code.
+- Run a DaemonSet that is deployed to the same nodes as the [CSI driver DaemonSet](https://github.com/openshift/secrets-store-csi-driver-operator/blob/main/assets/node.yaml).
+- Create a Unix Domain Socket under `/etc/kubernetes/secrets-store-csi-providers`, which the [CSI driver uses](https://github.com/openshift/secrets-store-csi-driver-operator/blob/d271705b2e56b20581dcc04514c9b307b67fbeec/assets/node.yaml#L39) to communicate with the provider.
```

Comment:
Create a Unix Domain Socket under `/etc/kubernetes/secrets-store-csi-providers` **on the host**.


BTW, do we want to spread usage of sockets in /etc? They belong to `/var`

Reply:
Updated. The CSI driver actually looks at both paths, but `/var/run/secrets-store-csi-providers` is the one that should be documented. The [Azure provider](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/dad67ff93e76554a489f72f2d4b0bef9afe2d417/deployment/provider-azure-installer.yaml#L72) we used in testing for example uses `/var`.